ElementX/Sources/Other/Logging/MXLog.swift

```diff
@@ -118,32 +118,57 @@ private var logger: SwiftyBeaver.Type = {
         logger.warning(message, file, function, line: line)
     }
     
-    public static func error(_ message: @autoclosure () -> Any,
+    /// Log error with additional details
+    ///
+    /// - Parameters:
+    ///     - message: Description of the error without any variables (this is to improve error aggregations by type)
+    ///     - context: Additional context-dependent details about the issue
+    public static func error(_ message: String,
                              _ file: String = #file,
                              _ function: String = #function,
                              line: Int = #line,
                              context: Any? = nil) {
-        logger.error(message(), file, function, line: line, context: context)
+        logger.error(message, file, function, line: line, context: context)
     }
     
     @available(swift, obsoleted: 5.4)
-    @objc public static func logError(_ message: String, file: String, function: String, line: Int) {
-        logger.error(message, file, function, line: line)
+    @objc public static func logError(_ message: String,
+                                      file: String,
+                                      function: String,
+                                      line: Int,
+                                      context: Any? = nil) {
+        logger.error(message, file, function, line: line, context: context)
     }
     
-    public static func failure(_ message: @autoclosure () -> Any,
+    /// Log failure with additional details
+    ///
+    /// A failure is any type of programming error which should never occur in production. In `DEBUG` configuration
+    /// any failure will raise `assertionFailure`
+    ///
+    /// - Parameters:
+    ///     - message: Description of the error without any variables (this is to improve error aggregations by type)
+    ///     - context: Additional context-dependent details about the issue
+    public static func failure(_ message: String,
                                _ file: String = #file,
                                _ function: String = #function,
```

Comment:
Not related to this PR, but seeing as we basically never enter these, maybe we should add the parameter names at some stage so its clear which default value you are replacing.
```suggestion
                               file: String = #file,
                               function: String = #function,
```